app/src/features/apiClient/screens/apiClient/components/modals/CopyAsModal/CopyAsModal.tsx

```diff
@@ -0,0 +1,84 @@
+import { RequestContentType, RQAPI } from "features/apiClient/types";
+import { Request as HarRequest } from "har-format";
+import { HTTPSnippet, availableTargets } from "@readme/httpsnippet";
+import { RQModal } from "lib/design-system/components";
+import { useCallback, useEffect, useState } from "react";
+import { apiRequestToHarRequestAdapter } from "../../../utils";
+import { Select } from "antd";
+import "./copyAsModal.scss";
+import CopyButton from "components/misc/CopyButton";
+
+interface CopyAsModalProps {
+  open: boolean;
+  onClose: () => void;
+  apiRequest: RQAPI.Request;
+}
+
+type TargetId = ReturnType<typeof availableTargets>[number]["key"];
+type ClientId = ReturnType<typeof availableTargets>[number]["clients"][number]["key"];
+
+const CopyAsModal = ({ apiRequest, onClose, open }: CopyAsModalProps) => {
+  const [harRequest, setHarRequest] = useState<HarRequest>();
+  const [snippetTypeId, setSnippetTypeId] = useState<string>("shell-curl");
+  useEffect(() => {
+    if (!apiRequest) {
+      return;
+    }
+
+    const harRequest = apiRequestToHarRequestAdapter(apiRequest);
+    setHarRequest(harRequest);
+  }, [apiRequest]);
+
+  const getSnippet = useCallback(() => {
+    if (!harRequest) {
+      return;
+    }
+
+    let snippet: any = "";
```

Comment:
Can we remove any type for snippet?
```suggestion
type SnippetResult = string | Error;
```